web/src/connectors/edit-model-connector/edit-model-connector.tsx

```diff
@@ -0,0 +1,195 @@
+            searchText: '',
+            sortConfig: { field: 'name', direction: SortingDirection.ASC },
+            filters: []
+        },
+        {}
+    );
+
+    const { data: basements } = useBasements(
+        {
+            page: 1,
+            size: 100,
+            searchText: '',
+            sortConfig: { field: 'name', direction: SortingDirection.ASC }
+        },
+        {}
+    );
+
+    const handleNext = () => {
+        setStepIndex(stepIndex + 1);
+    };
+    const handlePrev = () => {
+        setStepIndex(stepIndex - 1);
+    };
+    const finishButtonCaption = 'Add Model';
+
+    const renderForm = useCallback(
+        ({ lens, save }: IFormApi<ModelValues>) => {
+            const steps: WizardPropsStep[] = [
+                {
+                    title: 'Model',
+                    content: (
+                        <>
+                            <div className={wizardStyles['content__body']}>
+                                <EditModelSettings
+                                    lens={lens}
+                                    categories={categories?.data}
+                                    basements={basements?.data}
+                                    models={models?.data}
+                                />
+                            </div>
+                            <div className={wizardStyles['content__footer']}>
+                                {renderWizardButtons({
+                                    onNextClick: handleNext
+                                })}
+                            </div>
+                        </>
+                    )
+                },
+                {
+                    title: 'Data',
+                    content: (
+                        <>
+                            <div className={wizardStyles['content__body']}>
+                                <EditModelData lens={lens} />
+                            </div>
+                            <div className={wizardStyles['content__footer']}>
+                                {renderWizardButtons({
+                                    onPreviousClick: handlePrev,
+                                    onNextClick: () => {
+                                        save();
+                                    },
+                                    nextButtonCaption: finishButtonCaption
+                                })}
+                            </div>
+                        </>
+                    )
+                }
+            ];
+            return <Wizard steps={steps} returnUrl={MODELS_PAGE} stepIndex={stepIndex} />;
+        },
+        [stepIndex, basements]
+    );
+
+    const editModelMutation = useEditModelMutation();
+
+    const handleOnModelEdit = useCallback(async (values: ModelValues) => {
+        let categories: (string | number)[] = [];
+        if (values.jobs) {
+            values.jobs.forEach((el) => {
+                categories = [...categories, ...el.categories];
```

Comment:
I think it's better to use push here, to avoid creation of new array on each iteration